File: android/rctmgl/src/main/java-v10/com/mapbox/rctmgl/utils/LatLngBounds.java
package com.mapbox.rctmgl.utils;

import com.mapbox.geojson.Point;
import com.mapbox.maps.CoordinateBounds;

public class LatLngBounds {
  public static LatLngBounds from(double bbox, double bbox1, double bbox2, double bbox3) {
    return new LatLngBounds(bbox, bbox1, bbox2, bbox3);
  }

  LatLng getSouthWest() {
    return new LatLng(latSouth, lonWest);
  }

  LatLng getNorthEast() {
    return new LatLng(latNorth, lonEast);
  }

  LatLng[] toLatLngs() {
    return new LatLng[] {getNorthEast(), getSouthWest()};
  }
  
  double latNorth;
  double lonEast;
  double latSouth;
  double lonWest;

  LatLngBounds(double latNorth,double lonEast, double latSouth, double lonWest) {
    this.latNorth = latNorth;
    this.lonEast = lonEast;
    this.latSouth = latSouth;
    this.lonWest = lonWest;
  }

  public CoordinateBounds toBounds() {
    return new CoordinateBounds(
            Point.fromLngLat(lonWest, latSouth),
            Point.fromLngLat(lonEast, latNorth),
            false
    );
  }
}

File: android/src/main/java/com/rnmapbox/rnmbx/utils/LatLngBounds.kt
package com.rnmapbox.rnmbx.utils

import com.mapbox.geojson.Point
import com.mapbox.maps.CoordinateBounds
import com.rnmapbox.rnmbx.utils.LatLngBounds

class LatLngBounds internal constructor(
    var latNorth: Double,
    var lonEast: Double,
    var latSouth: Double,
    var lonWest: Double
) {
    val southWest: LatLng
        get() = LatLng(latSouth, lonWest)
    val northEast: LatLng
        get() = LatLng(latNorth, lonEast)

    fun toLatLngs(): Array<LatLng> {
        return arrayOf(northEast, southWest)
    }

    fun toBounds(): CoordinateBounds {
        return CoordinateBounds(
            Point.fromLngLat(lonWest, latSouth),
            Point.fromLngLat(lonEast, latNorth),
            false
        )
    }

    companion object {
        fun from(bbox: Double, bbox1: Double, bbox2: Double, bbox3: Double): LatLngBounds {
            return LatLngBounds(bbox, bbox1, bbox2, bbox3)
        }
    }
}